cleverpush/src/main/java/com/cleverpush/AddSubscriptionTags.java

```diff
@@ -0,0 +1,97 @@
+package com.cleverpush;
+
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.util.Log;
+
+import com.cleverpush.listener.AddTagCompletedListener;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class AddSubscriptionTags implements AddTagCompletedListener {
+
+    private String[] tagIds;
+    private String subscriptionId;
+    private String channelId;
+
+    public AddSubscriptionTags(String subscriptionId, String channelId, String... tagIds) {
+        this.subscriptionId = subscriptionId;
+        this.channelId = channelId;
+        this.tagIds = tagIds;
+    }
+
+    @Override
+    public void tagAdded(int currentPositionOfTagToAdd) {
+        if (currentPositionOfTagToAdd != tagIds.length - 1) {
+            addSubscriptionTag(tagIds[++currentPositionOfTagToAdd], this, currentPositionOfTagToAdd);
```

Comment:
if you do the suggestion below, you can turn the method to addSubscriptionTag(this, currentPositionOfTagToAdd++);